inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp

```diff
@@ -320,23 +321,47 @@ InferenceEngine::ExecutableNetworkInternal::Ptr
 Engine::LoadExeNetworkImpl(const InferenceEngine::CNNNetwork &network, const std::map<std::string, std::string> &config) {
     OV_ITT_SCOPED_TASK(itt::domains::MKLDNNPlugin, "Engine::LoadExeNetworkImpl");
 
+    static const std::vector<InferenceEngine::Precision> supportedInOutPrec = {
+        InferenceEngine::Precision::U8,
+        InferenceEngine::Precision::I8,
+        InferenceEngine::Precision::U16,
+        InferenceEngine::Precision::I16,
+        InferenceEngine::Precision::I32,
+        InferenceEngine::Precision::U64,
+        InferenceEngine::Precision::I64,
+        InferenceEngine::Precision::FP32,
+        InferenceEngine::Precision::BF16,
+        InferenceEngine::Precision::BOOL
+    };
+
     // verification of supported input
     InferenceEngine::InputsDataMap _networkInputs = network.getInputsInfo();
-    for (const auto &ii : _networkInputs) {
-        auto input_precision = ii.second->getPrecision();
-        if (input_precision != InferenceEngine::Precision::FP32 &&
-            input_precision != InferenceEngine::Precision::I32 &&
-            input_precision != InferenceEngine::Precision::U16 &&
-            input_precision != InferenceEngine::Precision::I16 &&
-            input_precision != InferenceEngine::Precision::I8 &&
-            input_precision != InferenceEngine::Precision::U8 &&
-            input_precision != InferenceEngine::Precision::BF16 &&
-            input_precision != InferenceEngine::Precision::BOOL &&
-            input_precision != InferenceEngine::Precision::I64 &&
-            input_precision != InferenceEngine::Precision::U64) {
+    for (const auto &in : _networkInputs) {
+        auto input_precision = in.second->getPrecision();
+        if (std::find(supportedInOutPrec.begin(), supportedInOutPrec.end(), input_precision) == supportedInOutPrec.end()) {
             THROW_IE_EXCEPTION << NOT_IMPLEMENTED_str
                                << "Input image format " << input_precision << " is not supported yet...";
         }
+
+        if (!MKLDNNExtensionUtils::isDefaultTensor(in.second->getTensorDesc())) {
+            THROW_IE_EXCEPTION << NOT_IMPLEMENTED_str
+                               << "CPU plug-in doesn't support network input with non default tensor descriptor. Input name: '" << in.first << "'";
```

Comment:
In general "non default tensor descriptor" is quite confusing for user. Maybe it will be better to throw smt like: "CPU plug-in doesn't support network inputs which are not dense or have non zero offsets"